src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -230,18 +230,33 @@ public void InitializeService(IProtocolEndpoint serviceHost)
                     if ((sqlConn != null) && !connInfo.IsSqlDW && !connInfo.IsAzure)
                     {
                         BackupOperation backupOperation = DisasterRecoveryService.Instance.SetBackupInput(helper.DataContainer, sqlConn, backupParams.BackupInfo);
+                        SqlTask sqlTask = null;
 
                         // create task metadata
                         TaskMetadata metadata = new TaskMetadata();
                         metadata.ServerName = connInfo.ConnectionDetails.ServerName;
-                        metadata.DatabaseName = connInfo.ConnectionDetails.DatabaseName;
-                        metadata.Name = SR.Backup_TaskName;
-                        metadata.IsCancelable = true;
+                        metadata.DatabaseName = connInfo.ConnectionDetails.DatabaseName;                        
                         metadata.Data = backupOperation;
-                        
-                        // create backup task and perform
-                        SqlTask sqlTask = SqlTaskManager.Instance.CreateTask(metadata, Instance.BackupTaskAsync);
-                        sqlTask.Run();
+
+                        if (backupParams.IsScripting)
+                        {
+                            metadata.Name = string.Format("{0} {1}", SR.BackupTaskName, SR.ScriptTaskName);
+                            metadata.TaskExecutionMode = TaskExecutionMode.Script;
+                            metadata.IsCancelable = false;
+                            sqlTask = SqlTaskManager.Instance.CreateTask(metadata, Instance.BackupScriptTaskAsync);
```

Comment:
I changed this method to get a cancelMthod also, if you use that, you can remove some of logic from you cancel method, it's up to you but might make the code cleaner